alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php

```diff
@@ -280,27 +289,32 @@ protected function serveEntriesAsPlaylist($entryIds, $durations, $referenceEntry
 		// build the media set
 		if ($isLive)
 		{
-			$mediaSet = $this->serveLivePlaylist($durations, $sequences,
-				$playlistStartTime, $firstClipStartTime, $initialClipIndex, $initialSegmentIndex, $origEntry->getRepeat());
+			$repeat = !!$origEntry->getRepeat(); // Convert to bool
+			$mediaSet = $this->serveLiveMediaSet($durations, $sequences,
+				$playlistStartTime, $firstClipStartTime, $initialClipIndex, $initialSegmentIndex, $repeat, !$repeat);
 		}
 		else
 		{
 			$mediaSet = array('durations' => $durations, 'sequences' => $sequences);
 		}
 
+		$this->sendJson($mediaSet, $storeCache, $isLive, $origEntry);
+	}
+
+	protected function sendJson($jsonArray, $storeCache, $isLive, $entry)
```

Comment:
If this code is used onyl by simulive flow - move it to simulaiveUtils

Reply:
This code is used also on serve playlist